datafusion/spark/src/function/datetime/date_add.rs

```diff
@@ -136,3 +145,48 @@ fn spark_date_add(args: &[ArrayRef]) -> Result<ArrayRef> {
     };
     Ok(Arc::new(result))
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use arrow::datatypes::Field;
+
+    #[test]
+    fn test_date_add_non_nullable_inputs() {
+        let func = SparkDateAdd::new();
+        let args = &[
+            Arc::new(Field::new("arg_int8", DataType::Int8, false)),
+            Arc::new(Field::new("arg_int16", DataType::Int16, false)),
+        ];
+
+        let ret_field = func
+            .return_field_from_args(ReturnFieldArgs {
+                arg_fields: args,
+                scalar_arguments: &[None, None],
+            })
+            .unwrap();
+
+        assert_eq!(ret_field.data_type(), &DataType::Date32);
+        assert!(!ret_field.is_nullable());
+    }
+
+    #[test]
+    fn test_date_add_nullable_inputs() {
+        let func = SparkDateAdd::new();
+        let args = &[
+            Arc::new(Field::new("arg_int8", DataType::Int8, false)),
+            Arc::new(Field::new("arg_int16", DataType::Int16, true)),
+            Arc::new(Field::new("arg_int32", DataType::Int32, false)),
+        ];
+
+        let ret_field = func
+            .return_field_from_args(ReturnFieldArgs {
+                arg_fields: args,
+                scalar_arguments: &[None, None, None],
+            })
+            .unwrap();
+
+        assert_eq!(ret_field.data_type(), &DataType::Date32);
+        assert!(ret_field.is_nullable());
+    }
+}
```

Comment:
Please also add a test for NULL at datafusion/sqllogictest/test_files/spark/datetime/date_add.slt if there is no such already